### PR TITLE
Fix docs build and update Zoom workflow

### PR DESCRIPTION
**Pull request by SimonHeybrock**
The Zoom notebook contained a mermaid diagram, which caused a Sphinx warning... and the notebook was never executed. Therefore, we did not notice that it was not refactored for the current Sciline release.

- Add mermaid extension.
- Update Zoom module and notebook to work with current Sciline.

### CODE REVIEW

File: src/ess/polarization/zoom.py [hunk 1]
```diff
@@ -65,78 +66,90 @@ def _get_time(dg: sc.DataGroup) -> sc.Variable:
...
     monitors = sc.concat(monitors, 'time')
     datetime = monitors.coords['datetime']
     monitors.coords['time'] = datetime - datetime.min()
     del monitors.coords['spectrum']
     del monitors.coords['detector_id']
-    monitors.variances = None
     return monitors
 
 
-def get_time_dependent_incident(
```
Comment (nvaytet) [inline src/ess/polarization/zoom.py:86]: Nice that we could now remove this :+1:

File: docs/user-guide/zoom.ipynb [hunk 2]
```diff
@@ -311,7 +292,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "func = pol_workflow.compute(pol.he3.He3TransmissionFunction[pol.Analyzer])"
+    "func = pol_workflow.compute(pol.TransmissionFunction[pol.Analyzer])"
    ]
   },
   {
```
Comment (nvaytet) [inline docs/user-guide/zoom.ipynb:298]: Changes look fine but it seems the notebook is not run in the documentation build (in the built docs from CI run at least).

Comment (SimonHeybrock) [reply docs/user-guide/zoom.ipynb:298]: Yes, sorry, I should have been clearer: The notebook requires Mantid, so it does not run in the actually docs build on CI.

Comment (nvaytet) [reply docs/user-guide/zoom.ipynb:298]: And I guess you don't want to install Mantid because you want to use `tox` and `pip`...

**Review by nvaytet — APPROVED** [review]

**Pull request merged**